### DISCUSSION TITLE
Allow output to Verified & Received folders with same filenames

### DISCUSSION THREAD

**Discussion by HermanEldering** [VerifyTests/Verify#597] [Ideas]
If the verified and received files would be split into two directories where the filenames inside both folders would be equal then it would be easier to use compare tools which can compare entire directories. 

From what I've read the verified and received parts are now hardcoded and the existing functionality to change filenames/paths can't support this.

**Comment by SimonCropp** [comment]
what approach are you currently using to accept/decline snapshots?
diffenginetray, resharper addin, rider addin, manually?

and what is your os and ide?

**Reply by HermanEldering** [reply]
I have just started looking into Verify, so I haven't settled on a workflow. But having one window open with a list/tree of all the changed files sounds nice to me, and that would work out of the box if folders can be compared.

That way I can also easily choose in which order I want to look at the diffs.

Currently I run the tests manually from VS, but for another project I'll want to integrate it in my build.

For development I use Windows 10 and Visual Studio 17.3. By default Verify opens VS in diff mode in the running VS instance, by adding my own registration I got VS to open in merge mode but that opens a new VS instance for each file (even with MDI enabled).

For source control merges I currently use kdiff3. Winmerge would work for me too. I'm not a fan of tooltray applications personally, especially as I will be using it sporadically in this case (depending on which project I work on).

**Reply by HermanEldering** [reply]
So as an example, after running Verify if there are any differences it could open a single instance of kdiff3 and it would look something like this. You'll see a list of the generated files with the changed files marked in red, in the same screen you will see the differences in the file you've selected and be able to accept the changes.

![image](https://cdn.kde.org/screenshots/kdiff3/ManualMerdge.png)

**Reply by SimonCropp** [reply]
> By default Verify opens VS in diff mode in the running VS instance, by adding my own registration I got VS to open in merge mode but that opens a new VS instance for each file (even with MDI enabled

yeah thats why VS is ranked last by order to use https://github.com/VerifyTests/DiffEngine#supported-tools

in fact if u have kdiff installed it should use that before VS

I dont think your proposal for "Allow output to Verified & Received folders with same filenames" has enough value to justify the complexity. a few problems:

 * A single test can produce multiple files and file formats. the best tool+compatible is selected for each file type. So having a single tool be able to do the diff for all the files doest map well for that.
 * when a test runs on multiple frameworks, the received file name includes the framework name. this is to prevent file access conflicts and also to allow the diff frameworks to produce their own (and possible different) received content
 * there is a workflow many people use where all the combos of parameterised tests should produce the same resulting verified file. so as with the multiple framework scenario, the received file names wont match the single verified file name
 * you would want to be able to compare the changes at a per test level. so achieve that you would need a directory per test to contain the  verified file(s). seems a fairly cubersom directory structure
